xds/internal/balancer/cdsbalancer/cdsbalancer.go

```diff
@@ -328,6 +329,8 @@ func (b *cdsBalancer) handleWatchUpdate(update *watchUpdate) {
 		return
 	}
 
+	client.SetMaxRequests(update.cds.ServiceName, update.cds.MaxRequests)
```

Comment:
Add a CDS test to check that counter is updated.

(Since you cannot read counter's fields, update max to small values, and `add`, and check return values)

Reply:
Made this change.